ert/data/_record.py

```diff
@@ -390,6 +392,31 @@ async def _load_blob_record(self) -> BlobRecord:
         return BlobRecord(data=self._record.data)
 
 
+class RecordTransformation:
+    async def file_to_bytes(location: pathlib.Path):
+        pass
+
+    async def bytes_to_file(data: bytes, location: pathlib.Path):
+        pass
+
+
+class RecordTarTransformation(RecordTransformation):
+    async def file_to_bytes(file_path: pathlib.Path) -> bytes:
+        tar_name = tempfile.NamedTemporaryFile().name
+        with tarfile.open(tar_name, "w") as tar:
+            for root, _, files in os.walk(file_path, topdown=False):
+                for file in files:
+                    tar.add(os.path.join(root, file))
+        with open(tar_name, "rb") as fb:
+            return fb.read()
+
+    async def bytes_to_file(data: bytes, location: pathlib.Path):
+        async with aiofiles.open(str(location) + ".tar", mode="wb") as ft:
+            await ft.write(data)
+        async with tarfile.open(str(location) + ".tar") as tar:
+            await tar.extractall(str(location))
+
+
```

Comment:
I think it would be a good time to introduce a new module inside `ert.data`, namely `_transformations`?

There are potential for cyclic imports when we start adding more transformations that aren't necessarily in the `ert.data._record` module.

Reply:
Planning to do that actually, but firstly wanted to get some feedback on it. Also not sure about whether the name as we might use more generally, ie. `RecordTransformation` ?